### PR TITLE
Refactor to stop inheriting from HashJoin

### PR DESCRIPTION
**Pull request by andygrove**
Signed-off-by: Andy Grove <andygrove@nvidia.com>

Move `GpuHashJoin` out of the shims and do not extend from Spark's `HashJoin`.

### CODE REVIEW

**Comment by andygrove** [comment]
build

File: sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuHashJoin.scala [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2019-2020, NVIDIA CORPORATION.
+ * Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
```
Comment (andygrove) [inline sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuHashJoin.scala:2]: Github is seeing this as a file move but I actually manually created this new file.

Comment (revans2) [reply sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuHashJoin.scala:2]: git looks for diffs and tries to guess what was a move and what was not.

**Comment by revans2** [comment]
Overall it looks good and I love that we deleted 700 lines of uneeded code.

**Comment by andygrove** [comment]
build

File: shims/spark301db/src/main/scala/com/nvidia/spark/rapids/shims/spark301db/GpuShuffledHashJoinExec.scala [hunk 2]
```diff
@@ -27,8 +27,9 @@ import org.apache.spark.sql.catalyst.optimizer.{BuildLeft, BuildRight, BuildSide
 import org.apache.spark.sql.catalyst.plans.JoinType
 import org.apache.spark.sql.catalyst.plans.physical.{Distribution, HashClusteredDistribution}
 import org.apache.spark.sql.execution.{BinaryExecNode, SparkPlan}
-import org.apache.spark.sql.execution.joins.ShuffledHashJoinExec
+import org.apache.spark.sql.execution.joins.{ShuffledHashJoinExec}
```
Comment (revans2) [inline shims/spark301db/src/main/scala/com/nvidia/spark/rapids/shims/spark301db/GpuShuffledHashJoinExec.scala:30]: nit don't need `{}` for a single import

**Review by revans2 — APPROVED** [review]
Just one nit. I am OK if we merge this as is.

File: shims/spark301db/src/main/scala/com/nvidia/spark/rapids/shims/spark301db/GpuShuffledHashJoinExec.scala (hunk 2 above)
Comment (andygrove) [reply shims/spark301db/src/main/scala/com/nvidia/spark/rapids/shims/spark301db/GpuShuffledHashJoinExec.scala:30]: Thanks. There are also a couple unused imports after this refactor so I will push a commit to clean this up.

**Comment by andygrove** [comment]
build

**Review by revans2 — APPROVED** [review]

**Pull request merged**